content/posts/why-we-choose-headless-components-for-cunningham/index.md

```diff
@@ -0,0 +1,245 @@
+you’ve lost all your investment like a car crashing in a wall in the middle of the highway.
+
+There is another solution were you can also contribute to the existing components library but the
+learning curve of large libraries like MUI is important and you are not guaranteed that your
+proposal will be accepted ( and that’s totally fair ).
+
+In some cases the missing feature you need might already be included into the paying version of the
+library, and adding those in a forked repository might cause some obvious issues regarding the
+commercial strategy of the library. Even if we wanted to buy the paying version, there would be some
+licensing issues because our code is licensed under MIT and the paying versions of those libraries
+is not, for obvious reasons.
+
+### The hard way
+
+And … why not coding our components from scratch? Just with plain Vanilla Javascript. Well,
+sometimes it’s fun to develop components from scratch, like turning raw materials into nice objects.
+
+Well, we had the constraint to ship a first version of a Data Table quite quickly, which can be an
+incredibly difficult subject considering numerous features and accessibility constraints. Creating
+such a component from scratch with lots of features can take weeks considering:
+
+- Pagination
+- Rows selection
+- Responsiveness
+- Resizable columns
+- Columns grouping
+- Sorting
+- Collapsible rows
+- Spanning
+- Virtualization
+- Etc ..
+
+.. And we’re not even talking about browser compatibilities.
+
+Creating hard components from scratch requires a lot of initial investment, but not only. It also
+requires a permanent investment to maintain those to browser updates and also to some exotic devices
+for instance. And let’s be honest, as developers our first job is to NOT reinvent the wheel.
+
+We though there might be another approach that can offer the perfect balance between customization
+and development efforts.
+
+### The way of the cunning
+
+And what if there is middle approach more flexible than library components but without the chore
+that represent to rewrite all components from scratch?
+
+It’s time to talk about Headless components!
+
+Headless components are a type of library that provide logic, state, processing and API for UI
+elements and interactions but do not provide markup, styles or pre-built implementations. This means
+that it is possible to create a fully featured and accessible data table without having to start
+from scratch and spending months to develop it. Headless component libraries, like react-table,
+offer all the features that you would expect from a data table, so that you can focus solely on the
+[“front-of-the-front” without having to worry about the “back-of-the-front”](https://bradfrost.com/blog/post/front-of-the-front-end-and-back-of-the-front-end-web-development/)[.](https://www.notion.so/Magnify-Packages-front-7f714dad07074ef484f15d3fa153b4b4)
```

Comment:
The https://www.notion.so/Magnify-Packages-front-7f714dad07074ef484f15d3fa153b4b4 URL is not public, right? What's its purpose?

Reply:
Don't know why I tried to hide a link into this point 😅